File: apps/party-time-backend/src/main/java/com/partytime/jpa/entity/EventParticipant.java
package com.partytime.jpa.entity;

import com.partytime.jpa.DatabaseConstants;
import com.partytime.jpa.factory.BaseEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Entity
@Table(name = DatabaseConstants.EventParticipant.TABLE_NAME)
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class EventParticipant extends BaseEntity<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @ManyToOne
    @JoinColumn(name = DatabaseConstants.EventParticipant.COLUMN_ACCOUNT_ID)
    private Account account;

    @NotNull
    @ManyToOne
    @JoinColumn(name = DatabaseConstants.EventParticipant.COLUMN_EVENT_ID)
    private Event event;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(name = DatabaseConstants.EventParticipant.COLUMN_STATUS)
    private Status status;

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/entity/EventParticipant.kt
package com.partytime.jpa.entity

import com.partytime.jpa.DatabaseConstants
import com.partytime.jpa.factory.EntityWithLongId
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = DatabaseConstants.EventParticipant.TABLE_NAME)
class EventParticipant(
    @ManyToOne
    @JoinColumn(name = DatabaseConstants.EventParticipant.COLUMN_ACCOUNT_ID)
    var account: Account,
    @ManyToOne
    @JoinColumn(name = DatabaseConstants.EventParticipant.COLUMN_EVENT_ID)
    var event: Event,
    @Enumerated(EnumType.STRING)
    @Column(name = DatabaseConstants.EventParticipant.COLUMN_STATUS)
    var status: Status
): EntityWithLongId()
